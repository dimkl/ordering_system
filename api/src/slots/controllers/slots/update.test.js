/**
 * @integration-test true
 * @data-factory true
 */
describe("PATCH /slots/:slot_id", () => {
  let knex;
  beforeAll(() => knex = require('../../../shared/setupModels')());
  beforeEach(() => knex.raw('truncate users, time_slots, slots, sections, shops cascade;'));
  afterAll(() => knex.destroy());

  it("updates and returns updated slot", async () => {
    const slot = await DataFactory.createSlot();
    const section = await DataFactory.createSection({}, {}, { email: 'aloha@example.com' });

    const response = await request.patch(`/slots/${slot.uuid}`)
      .send({
        active: false,
        user_id: section.user.uuid,
        section_id: section.uuid
      })
      .set("Accept", "application/json");

    expect(response.status).toBe(200);
    expect(response.body).toMatchObject({
      "active": false,
      "created_at": slot.created_at.toISOString(),
      "updated_at": expect.any(String),
      "id": slot.id,
      "section_id": section.uuid,
      "sku": "table-1",
      "user_id": section.user.uuid,
      "uuid": slot.uuid
    });
    expect(slot.user_id).not.toEqual(section.user.id);
    expect(slot.section_id).not.toEqual(section.id);
    expect(slot.active).not.toEqual(false);
    const unixUpdatedAt = new Date(response.body.updated_at).getTime();
    expect(unixUpdatedAt).toBeGreaterThan(slot.updated_at.getTime());
  });

  it("does not have required properties", async () => {
    const slot = await DataFactory.createSlot();

    const response = await request.patch(`/slots/${slot.uuid}`)
      .send({})
      .set("Accept", "application/json");

    expect(response.status).toBe(200);
    expect(response.body).toMatchObject({
      "active": true,
      "created_at": slot.created_at.toISOString(),
      "updated_at": slot.updated_at.toISOString(),
      "id": slot.id,
      "section_id": slot.section.uuid,
      "sku": "table-1",
      "user_id": slot.user.uuid,
      "uuid": slot.uuid
    });
  });

  it("throws validation error with additional properties", async () => {
    const slot = await DataFactory.createSlot();

    const response = await request.patch(`/slots/${slot.uuid}`)
      .send({ created_at: Date.now() })
      .set("Accept", "application/json");

    expect(response.status).toBe(400);
    expect(response.body).toMatchSnapshot();
  });
});