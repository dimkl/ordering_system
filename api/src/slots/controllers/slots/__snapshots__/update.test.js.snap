// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PATCH /slots/:slot_id throws validation error with additional properties 1`] = `
Array [
  Object {
    "instancePath": "",
    "keyword": "additionalProperties",
    "message": "must NOT have additional properties",
    "params": Object {
      "additionalProperty": "created_at",
    },
    "schemaPath": "#/additionalProperties",
  },
]
`;
